tidy3d/components/medium.py

```diff
@@ -1321,6 +1359,60 @@ def _not_loaded(field):
         elif isinstance(field, UnstructuredGridDataset):
             return any(len(subfield) == 0 for subfield in [field.points, field.cells, field.values])
 
+    def _derivative_field_cmp(
+        self,
+        E_der_map: ElectromagneticFieldDataset,
+        eps_data: PermittivityDataset,
+        dim: str,
+    ) -> np.ndarray:
+        coords_interp = {key: val for key, val in eps_data.coords.items() if len(val) > 1}
+        dims_sum = {dim for dim in eps_data.coords.keys() if dim not in coords_interp}
```

Comment:
Writing mostly so I can comment on dim sum 😂 Maybe style: `dims_sum = set(eps_data.coords.keys()) - set(coords_interp)` ... the comprehension might be more accessible though so idk.

Reply:
haha